--- mp3c.c	Wed May 17 21:37:48 2000
+++ ../../../mp3sb_client/curses/mp3c.c	Sun Jul 23 01:38:23 2000
@@ -10,6 +10,8 @@
 #include <termios.h>
 #include <sys/time.h>
 #include <sys/ioctl.h>
+#include <sys/types.h>
+#include <pwd.h>
 
 #include <sys/socket.h>
 #include <netinet/in.h>
@@ -19,7 +21,7 @@
 #include <ncurses.h>
 #include <menu.h>
 
 /* --------------------------------------------------------------------- */
 /* global variables                                                      */
@@ -776,6 +778,16 @@
     len = sizeof(addr);
     if (-1 == connect(mp3,(struct sockaddr*)&addr,len))
 	panic("connect(%s:2345): %s",mp3server,strerror(errno));
+
+    /* advertise client version & username */
+    { 
+      uid_t self = getuid();
+      struct passwd *pass = getpwuid(self);
+      
+      write(mp3,"HELLO mp3c ",11);
+      write(mp3,pass->pw_name,strlen(pass->pw_name));
+      write(mp3,"\r\n",2);
+    }
 
     /* command mode */
     if (optind < argc) {
